redis/src/main/scala/zio/redis/CommandExecutor.scala

```diff
@@ -0,0 +1,9 @@
+package zio.redis
+
+import zio.schema.codec.BinaryCodec
+
+trait CommandExecutor {
+  implicit def codec: BinaryCodec
+
+  implicit def executor: RedisExecutor
+}
```

Comment:
Let's rename it somehow, as this is not really an executor.

Reply:
yeah i was struggling with naming this one, considering those two methods were making Redis trait before this change. could it be something like `RedisEnvironment` or `CommandExecutable`. Any suggestion?

Reply:
Environment sounds good. They don't need to be implicit anymore either. The question remains: can we remove it completely?

Reply:
i don't think we can, unless we want to split command creation and execution. If we changed API traits to return `RedisCommand` and pushed invocation of `run` then we wouldn't need these dependencies on those traits. Or do you have some ideas?

Reply:
I would not "expose" run. As mentioned below, 2nd parameter group is a bit too much as well, imo. On the other side, it makes sense if we restore implicits. :thinking:

Reply:
That's not bad. We might be able to combine it with `returning` somehow and make a nice "proxy", but let's leave it for the follow-up PR (if you're willing to dig into it :) ).